src/mempool.rs

```diff
@@ -0,0 +1,156 @@
+use crate::{
+    block::Block, burnfee::BurnFee, keypair::Keypair, time::create_timestamp,
+    transaction::Transaction,
+};
+
+pub const GENESIS_PERIOD: u64 = 21500;
+
+/// The `Mempool` is the structure that collects blocks and transactions
+/// and is control of discerning whether the node is allowed to create a block.
+/// It bundles the block, the sends it to `Blockchain` to be added to the longest chain.
+/// New `Block`s coming in over the network will hit the `Mempool` before being added to
+/// the `Blockchain`
+pub struct Mempool {
+    /// `Keypair` used to sign and bundle blocks
+    _keypair: Keypair,
+    /// A list of blocks to be added to the longest chain
+    _blocks: Vec<Block>,
+    /// A list of `Transaction`s to be bundled into `Block`s
+    _transactions: Vec<Transaction>,
+    ///
+    _burnfee: BurnFee,
+    /// The current work available in fees found in the list of `Transaction`s
+    work_available: u64,
+}
+
+impl Mempool {
+    /// Creates new `Memppol`
+    pub fn new(keypair: Keypair) -> Self {
+        Mempool {
+            _keypair: keypair,
+            _burnfee: BurnFee::new(10.0),
+            _blocks: vec![],
+            _transactions: vec![],
+            work_available: 0,
+        }
+    }
+
+    pub fn work_available(&self) -> u64 {
+        self.work_available
+    }
+
+    /// Attempt to create a new `Block`
+    ///
+    /// * `previous_block` - `Option` of previous `Block`
+    fn _try_bundle(&mut self, previous_block: Option<Block>) -> Option<Block> {
+        if self._can_bundle_block(previous_block) {
+            Some(self._bundle_block(None))
```

Comment:
Shouldn't you pass the previous block instead of `None`?

Reply:
oops, yes it should! *facepalm*